### PR TITLE
[bsp][ch32v307] 添加can驱动

### PR DESCRIPTION
**Pull request by cazure**
## 拉取/合并请求描述：(PR description)

[
CH32V307 添加can驱动。
CAN1和CAN2均可以使用，而且调整好波特率
]

以下的内容不应该在提交PR时的message修改，修改下述message，PR会被直接关闭。请在提交PR后，浏览器查看PR并对以下检查项逐项check，没问题后逐条在页面上打钩。
The following content must not be changed in the submitted PR message. Otherwise, the PR will be closed immediately. After submitted PR, please use a web browser to visit PR, and check items one by one, and ticked them if no problem.

### 当前拉取/合并请求的状态 Intent for your PR

必须选择一项 Choose one (Mandatory):

- [ ] 本拉取/合并请求是一个草稿版本 This PR is for a code-review and is intended to get feedback
- [x] 本拉取/合并请求是一个成熟版本 This PR is mature, and ready to be integrated into the repo

### 代码质量 Code Quality：

我在这个拉取/合并请求中已经考虑了 As part of this pull request, I've considered the following:

- [x] 已经仔细查看过代码改动的对比 Already check the difference between PR and old code
- [x] 代码风格正确，包括缩进空格，命名及其他风格 Style guide is adhered to, including spacing, naming and other styles
- [x] 没有垃圾代码，代码尽量精简，不包含`#if 0`代码，不包含已经被注释了的代码 All redundant code is removed and cleaned up
- [x] 所有变更均有原因及合理的，并且不会影响到其他软件组件代码或BSP All modifications are justified and not affect other components or BSP
- [x] 对难懂代码均提供对应的注释 I've commented appropriately where code is tricky
- [x] 本拉取/合并请求代码是高质量的 Code in this PR is of high quality
- [x] 本拉取/合并使用[formatting](https://github.com/mysterywolf/formatting)等源码格式化工具确保格式符合[RT-Thread代码规范](../documentation/contribution_guide/coding_style_cn.md) This PR complies with [RT-Thread code specification](../documentation/contribution_guide/coding_style_en.txt)

### CODE REVIEW

File: bsp/wch/risc-v/ch32v307v-r1/board/Kconfig [hunk 1]
```diff
@@ -152,7 +152,20 @@ menu "On-chip Peripheral Drivers"
...
+        default n
+        select RT_USING_CAN
+        if BSP_USING_CAN
+            config BSP_USING_CAN1
+                bool "using CAN1"
+                default n
+            config BSP_USING_CAN2
+                bool "using CAN2"
+                default n
+        endif
```
Comment (mysterywolf) [inline bsp/wch/risc-v/ch32v307v-r1/board/Kconfig:167]: 这个要有缩进

Comment (mysterywolf) [inline bsp/wch/risc-v/ch32v307v-r1/board/Kconfig:167]:
```suggestion
	menuconfig BSP_USING_CAN
            bool "Enable CAN"
            default n
            select RT_USING_CAN
            if BSP_USING_CAN
                config BSP_USING_CAN1
                    bool "using CAN1"
                    default n
                config BSP_USING_CAN2
                    bool "using CAN2"
                    default n
            endif
```

**Review by mysterywolf — APPROVED** [review]

File: bsp/wch/risc-v/Libraries/ch32_drivers/drv_can.h [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+#define __DRV_CAN_H__
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+#include <board.h>
+#include <rtdevice.h>
+#include <rtthread.h>
```
Comment (mysterywolf) [inline bsp/wch/risc-v/Libraries/ch32_drivers/drv_can.h:21]: 这几个头文件的定义放在.c里

File: bsp/wch/risc-v/Libraries/ch32_drivers/drv_can.c [hunk 3]
```diff
@@ -0,0 +1,1165 @@
...
+#include <board.h>
+#include <rtdevice.h>
+#include "drv_can.h"
+
+#define DBG_TAG "drv_can"
+//#define DBG_LVL DBG_INFO
+#define DBG_LVL DBG_LOG
+#define DBG_ENABLE
+#include <rtdbg.h>
+//#include <drv_log.h>
```
Comment (mysterywolf) [inline bsp/wch/risc-v/Libraries/ch32_drivers/drv_can.c:24]: 这块再确认一下

**Review by mysterywolf — CHANGES_REQUESTED** [review]

File: bsp/wch/risc-v/Libraries/ch32_drivers/drv_can.c (hunk 3 above)
Comment (cazure) [reply bsp/wch/risc-v/Libraries/ch32_drivers/drv_can.c]: 重新改了下头文件位置和debug等级

**Review by mysterywolf — APPROVED** [review]

**Review by Guozhanxin — APPROVED** [review]

**Pull request merged**